virtualizarr/readers/dmrpp.py

```diff
@@ -406,24 +408,33 @@ def _parse_variable(self, var_tag: ET.Element) -> Variable:
                 chunks_shape = shape
             chunkmanifest = self._parse_chunks(chunks_tag, chunks_shape)
             # Filters
-            filters = self._parse_filters(chunks_tag, dtype)
+            codecs = self._parse_filters(chunks_tag, dtype)
         # Attributes
         attrs: dict[str, Any] = {}
         for attr_tag in var_tag.iterfind("dap:Attribute", self._NS):
             attrs.update(self._parse_attribute(attr_tag))
         # Fill value is placed in zarr array's fill_value and variable encoding and removed from attributes
         encoding = {k: attrs.get(k) for k in self._ENCODING_KEYS if k in attrs}
         fill_value = attrs.pop("_FillValue", None)
-        # create ManifestArray and ZArray
-        zarray = ZArray(
-            chunks=chunks_shape,
-            dtype=dtype,
-            fill_value=fill_value,
-            filters=filters,
-            order="C",
+        # create ManifestArray
+        metadata = ArrayV3Metadata(
```

Comment:
This call seems to have quite a lot of boilerplate that authors of virtualizarr readers will never need, so as it's generally useful perhaps we should provide our own constructor function in `.utils`?

Reply:
Updated in https://github.com/zarr-developers/VirtualiZarr/pull/429/commits/86d1de5af91501fbdd0b0d8e2f8545645993a7d1